fastchat/serve/openai_api_server.py

```diff
@@ -269,6 +271,39 @@ async def show_available_models():
     return ModelList(data=model_cards)
 
 
+# TODO: Have check_length and count_tokens share code.
+@app.post("/v1/token_check")
+async def count_tokens(request: TokenCheckRequest):
+    """Checks the token count against your message"""
+    async with httpx.AsyncClient() as client:
+        worker_addr = await _get_worker_address(request.model, client)
+
+        response = await client.post(
+            worker_addr + "/model_details",
+            headers=headers,
+            json={},
+            timeout=WORKER_API_TIMEOUT,
+        )
+        context_len = response.json()["context_length"]
+
+        response = await client.post(
+            worker_addr + "/count_token",
+            headers=headers,
+            json={"prompt": request.prompt},
+            timeout=WORKER_API_TIMEOUT,
+        )
+        token_num = response.json()["count"]
+
+    # TODO: Fix this for other models
+    context_len = 2048
```

Comment:
Use the returned cntext_len instead of 2048

Reply:
Makes sense, I was only copying "check_length" but it looks like this'll work in both places, should I change it for both?

Reply:
I have fixed that in other PRs